contribs/src/main/java/com/netflix/conductor/contribs/queue/shotgun/ShotgunQueue.java

```diff
@@ -250,7 +234,12 @@ private void onMessage(Subscription subscription, ShotgunOuterClass.Message mess
         dstMsg.setPayload(payload);
         dstMsg.setReceived(System.currentTimeMillis());
 
-        logger.info(String.format("Received message for %s %s=%s", subscription.getSubject(), dstMsg.getId(), payload));
+        NDC.push("event-"+dstMsg.getId());
```

Comment:
Set NDC value (owner) so the 'Received ...' log message will be associated to the same owner when event being processed